### ISSUE TITLE
Separate FFT size from buffer size

### ISSUE THREAD

**Issue by nevosegal** [meyda/meyda#261]
At the moment, the user can't set an FFT size that is different from the input buffer size. We should add this option, and default the FFT size value to be equal to the buffer size.

**Comment by hughrawlinson** [comment]
Would this need to handle zero-padding and/or resampling?

**Comment by nevosegal** [comment]
No resampling, yes zero padding.

**Comment by hughrawlinson** [comment]
some logic like:

```js
if (fftSize != bufferSize) {
  zeroPad(signal, zeroPaddingMode);
}
```

?

That would allow the bufferSize to be non-power-of-two, which could be handy. Would we also have to have a prefix/postfix/split mode for letting people choose where the zeros go? Or is 'half before and half after' the same as 'all before' or 'all after'? Would we always pad just up to the next power of two, or would we allow people to specify that they want padding up to the 2nd next power of two, etc?

**Comment by nevosegal** [comment]
I don't think it matters where you add the zeroes. I often pad equally on both sides, but I've seen people pad after the audio data. The output won't change, so I don't think we need to provide the users with `mode` option.

We do need to make sure that we zero-pad after we apply the windowing function on the audio data. We don't want window the zero-padded input.

**Comment by nevosegal** [comment]
And yeah, implementing zero-padding opens the door to allowing non-power-of-two input, which could be awesome (especially for the offline mode).

So:
- If the input buffer size is non-power-of-two, and the user doesn't specify an FFT size, we should pad the input to the next closest power-of-two. For example, input buffer size of 500 should be zero-padded to be of length 512.
- If the user _does_ specify an FFT size, we should:
    - Only accept power-of-two FFT sizes.
    - Make sure the FFT size isn't smaller than the buffer size.
    - Pad the input buffer to the specified FFT size.

**Comment by hughrawlinson** [comment]
Awesome!

**Comment by relativeflux** [comment]
Any update on this - it would be a great feature.

**Comment by hughrawlinson** [comment]
I agree! This ticket is open for whoever has time to submit a PR, are you interested @relativeflux?
